### ISSUE TITLE
Wrong step definition created when having "("

### ISSUE THREAD

**Issue by aslakhellesoy** [cucumber/cucumber-ruby#1291]
_From @christianbaumann on April 19, 2018 9:43_

## Summary

I have a step definition which contains "(" and ")", eg. 
> And the room information (room code and availability) is correct

## Expected Behavior

When I run cucumber and have no step definition defined, I would expect a matching step definition to be defined
> And(/^the room information \(room code and availability\) is correct$/) do

## Current Behavior

Instead, following step definition is created:
> And("the room information \(room code and availability) is correct") do

## Your Environment

* Ruby: ruby 2.3.3p222 (2016-11-21 revision 56859) [x64-mingw32]
* Cucumber:
cucumber (3.1.0)
cucumber-core (3.1.0)
cucumber-expressions (5.0.17, 5.0.15)
cucumber-tag_expressions (1.1.1)
cucumber-wire (0.0.1)
* Operating System and version: Windows 8.1 Enterprise

_Copied from original issue: cucumber/cucumber#364_

**Comment by aslakhellesoy** [comment]
As of [3.0.0.pre.2](https://github.com/cucumber/cucumber-ruby/blob/master/CHANGELOG.md#300pre2), cucumber-ruby supports both [Cucumber Expressions](https://cucumber.io/blog/2017/07/26/announcing-cucumber-expressions) and Regular Expressions.

Generated snippets use Cucumber Expressions, not Regular Expressions.

I think you may have stumbled upon #333 which should have been fixed by #334, but it looks like there might still be a bug with it.

This needs more investigation....

**Comment by aslakhellesoy** [comment]
@christianbaumann here is a workaround for you. The generated snippet currently is:

```ruby
Given("the room information \(room code and availability) is correct") do
  pending # Write code here that turns the phrase above into concrete actions
end
```

Either change double quotes to single quotes:

```ruby
Given('the room information \(room code and availability) is correct') do
  pending # Write code here that turns the phrase above into concrete actions
end
```

Or double-escape the opening parenthesis:

```ruby
Given("the room information \\(room code and availability) is correct") do
  pending # Write code here that turns the phrase above into concrete actions
end
```

The bug is in cucumber-ruby, and not in cucumber-expressions. We should change the snippet generator to use single quotes rather than double quotes, and only escape single quotes in the expressions itself.

**Comment by danascheider** [comment]
I'm keeping this open for now but may close it soon and create another PR that uses the double escape instead of single quotes.

**Comment by xtrasimplicity** [comment]
Bumping to remove the tag. :)

**Comment by danascheider** [comment]
Sorry for the delay on this! It's still on my radar but I'm in the process of moving to Australia so things are particularly hectic atm.

**Comment by xtrasimplicity** [comment]
Awesome! All the best with the move! :)

**Comment by danascheider** [comment]
Question on this @aslakhellesoy (then I'll stop tagging you in stuff for now!): Do we want to accommodate curly braces in step definitions? We could implement this such that Cucumber first looks for a param type with the name in the curly braces and, if it doesn't find one, assumes the curly braces are part of the step. I can think of good reasons for or against this, so I thought I'd ask for your input. @brasmusson, @mattwynne, @tooky - maybe you guys have thoughts on this as well?

**Comment by aslakhellesoy** [comment]
@danascheider I think this bug was (incidentally?) fixed by @acant in #1323.

No need to handle curly braces in a special way - cucumber-expressions handles escaping.

**Comment by danascheider** [comment]
Sweet!

**Comment by aslakhellesoy** [comment]
Fixed by #1323
